### PR TITLE
Add support for buffers

### PR DESCRIPTION
**Pull request by kevva**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,8 +1,24 @@
...
+	var start = buf.slice(0, i);
+	var end = replaceBuf(buf.slice(i + crlf.length));
+	var len = i + start.length + end.length;
+
+	return Buffer.concat([start, '\n', end], len);
+};
+
+module.exports = function (x) {
+	if (typeof x !== 'string' && !Buffer.isBuffer(x)) {
+		throw new TypeError('Expected a string or a buffer');
```
Comment (SamVerschueren) [inline index.js]:
``` js
throw new TypeError('Expected a `string` or a `Buffer`, got `' + (typeof x) + '`');
```

Not sure on the `Buffer` capitalization though.

File: index.js [hunk 2]
```diff
@@ -1,8 +1,24 @@
...
 	}
 
-	return str.replace(/\r\n/g, '\n');
+	return Buffer.isBuffer(x) ? replaceBuf(x) : x.replace(/\r\n/g, '\n');
```
Comment (SamVerschueren) [inline index.js]: Maybe add `var CRLF = '\r\n';` at the top and reuse it here with `x.replace(new RegExp(CRLF, 'g'), '\n')` and in the `replaceBuf` function?

File: index.js (hunk 1 above)
Comment (sindresorhus) [reply index.js]: That's fine. I usually capitalize all non-primitives.

Comment (sindresorhus) [reply index.js]: Unrelated, but I've been thinking about doing:

``` js
throw new TypeError('Expected `string` or `Buffer`, got `' + x + '` (' + (typeof x) + ')');
```

Result:

```
TypeError: Expected `string` or `Buffer`, got `foo` (string)
```

That way the user can see the actual value too. Also dropped the `a`'s. What do you guys think?

Comment (SamVerschueren) [reply index.js]: Should we call `x.toString()` or not? Just wondering what will happen if large objects/values/arrays are being passed inside the method.

> Also dropped the a's.

👍

Comment (sindresorhus) [reply index.js]: Ah yeah. That's a good point. Maybe just the type then. Simpler.

File: index.js (hunk 2 above)
Comment (kevva) [reply index.js]: :+1:

**Comment by SamVerschueren** [comment]
Might be a good idea to also test the `throw new TypeError` path if no `string` or `Buffer` is provided?

Also an API section might be a nice addition.

**Comment by sindresorhus** [comment]
Good stuff :)

**Comment by bezoerb** [comment]
Thanks

**Pull request merged**